webbpsf/distortion.py

```diff
@@ -40,8 +42,150 @@ def _get_default_siaf(instrument, aper_name):
 
     return aper
 
+# Functions for applying distortion from SIAF polynomials
+def distort_image(hdulist_or_filename, ext=0, to_frame='sci', fill_value=0, 
+                  xnew_coords=None, ynew_coords=None, return_coords=False,
+                  aper=None):
+    """ Distort an image
 
-# Function for applying distortion from SIAF polynomials
+    Apply SIAF instrument distortion to an image that is assumed to be in 
+    its ideal coordinates. The header information should contain the relevant
+    SIAF point information, such as SI instrument, aperture name, pixel scale,
+    detector oversampling, and detector position ('sci' coords).
+
+    This function then transforms the image to the new coordinate system using
+    scipy's RegularGridInterpolator (linear interpolation).
+
+    Parameters
+    ----------
+    hdulist_or_filename : str or HDUList
+        A PSF from WebbPSF, either as an HDUlist object or as a filename
+    ext : int
+        Extension of HDUList to perform distortion on.
+    fill_value : float or None
+        Value used to fill in any blank space by the skewed PSF. Default = 0.
+        If set to None, values outside the domain are extrapolated.
+    to_frame : str
+        Type of input coordinates. 
+
+            * 'tel': arcsecs V2,V3
+            * 'sci': pixels, in conventional DMS axes orientation
+            * 'det': pixels, in raw detector read out axes orientation
+            * 'idl': arcsecs relative to aperture reference location.
+
+    xnew_coords : None or ndarray
+        Array of x-values in new coordinate frame to interpolate onto.
+        Can be a 1-dimensional array of unique values, in which case 
+        the final image will be of size (ny_new, nx_new). Or a 2d array 
+        that corresponds to full regular grid and has same shape as 
+        `ynew_coords` (ny_new, nx_new). If set to None, then final image
+        is same size as input image, and coordinate grid spans the min
+        and max values of siaf_ap.convert(xidl,yidl,'idl',to_frame). 
+    ynew_coords : None or ndarray
+        Array of y-values in new coordinate frame to interpolate onto.
+        Can be a 1-dimensional array of unique values, in which case 
+        the final image will be of size (ny_new, nx_new). Or a 2d array 
+        that corresponds to full regular grid and has same shape as 
+        `xnew_coords` (ny_new, nx_new). If set to None, then final image
+        is same size as input image, and coordinate grid spans the min
+        and max values of siaf_ap.convert(xidl,yidl,'idl',to_frame). 
```

Comment:
What are the use cases for passing in coordinate arrays here? I can see this adds flexibility, but also complexity. I'm curious if you have particular needs or scenarios in mind that will use this.

Reply:
Useful for plotting purposes onto custom grid spacing. I am currently using this for visualizing model images onto already existing V2/V3 coordinate grids. If you want, I can remove these options here, since I've actually already made  a slightly different version of this function for `webbpsf_ext`.

Reply:
No, definitely don't remove. I could tell you had put in the option for some reason, and wanted to understand what it was.